src/routes/reports/reports.jsx

```diff
@@ -1,11 +1,18 @@
 import classNames from "classnames";
 import styles from "./reports.module.scss";
 import { OpenPosition, Statements } from "../../components";
-import { Show, createSignal } from "solid-js";
+import { Show, createSignal, onMount } from "solid-js";
+import { login_information } from "Stores/base-store";
+import { loginUrl } from "Constants/deriv-urls";
 
 const Reports = () => {
   const [tab_index, setTabIndex] = createSignal(1);
 
+  onMount(() => {
+    if (!login_information.is_logged_in) {
```

Comment:
Can this particular block (To redirect user to login) be created as a utility function, so it can be reused in all route navigation landing pages?